File: frontend/src/app/features/support/services/support-api.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../../environments/environment';
import { SupportConversationSummary, SupportMessage } from '../models/support.model';

@Injectable({ providedIn: 'root' })
export class SupportApiService {
  private readonly http = inject(HttpClient);
  private readonly apiUrl = `${environment.apiBaseUrl}/support`;

  getMyChat(): Observable<SupportMessage[]> {
    return this.http.get<SupportMessage[]>(`${this.apiUrl}/my-chat`);
  }

  getConversations(): Observable<SupportConversationSummary[]> {
    return this.http.get<SupportConversationSummary[]>(`${this.apiUrl}/conversations`);
  }

  getConversation(userId: number): Observable<SupportMessage[]> {
    return this.http.get<SupportMessage[]>(`${this.apiUrl}/conversations/${userId}`);
  }

  sendMessage(content: string, receiverId?: number | null): Observable<SupportMessage> {
    return this.http.post<SupportMessage>(`${this.apiUrl}/messages`, {
      receiverId: receiverId ?? null,
      content,
    });
  }
}

File: mobile/app/src/main/java/com/drumigo/mobile/data/api/SupportApiService.java
package com.drumigo.mobile.data.api;

import com.drumigo.mobile.data.model.support.SupportConversationSummaryResponse;
import com.drumigo.mobile.data.model.support.SupportMessageCreateRequest;
import com.drumigo.mobile.data.model.support.SupportMessageResponse;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

/**
 * Support chat API. Base URL is API_BASE_URL (e.g. .../api).
 */
public interface SupportApiService {

    @GET("support/my-chat")
    Call<List<SupportMessageResponse>> getMyChat();

    @GET("support/conversations")
    Call<List<SupportConversationSummaryResponse>> getConversations();

    @GET("support/conversations/{userId}")
    Call<List<SupportMessageResponse>> getConversation(@Path("userId") long userId);

    @POST("support/messages")
    Call<SupportMessageResponse> sendMessage(@Body SupportMessageCreateRequest request);
}
